csharp/BYOAuthFlow/Program.cs

```
@@ -0,0 +1,228 @@
+            }).First(d => d.Id == deploymentId);
+        }
+
+        private static string CreateLoginTokenForDeployment(LoginTokenServiceClient ltClient,
+            string pit, string deploymentId)
+        {
+            Console.WriteLine("Generate a Login Token for the selected deployment");
+            var createLoginTokenResponse = ltClient.CreateLoginToken(
+                new CreateLoginTokenRequest
+                {
+                    PlayerIdentityToken = pit,
+                    DeploymentId = deploymentId,
+                    LifetimeDuration = Duration.FromTimeSpan(new TimeSpan(0, 0, 30, 0)),
+                    WorkerType = "UnityClient",
+                });
+            var loginToken = createLoginTokenResponse.LoginToken;
+            return loginToken;
+        }
+
+        private static void ConnectToLocator(string locatorHost, ushort locatorPort, bool insecureConnection, string pit, string loginToken)
+        {
+            Console.WriteLine("Connect to the deployment using the Login Token and PIT");
+            var locatorParameters = new LocatorParameters
+            {
+                PlayerIdentity = new PlayerIdentityCredentials
+                {
+                    PlayerIdentityToken = pit,
+                    LoginToken = loginToken,
+                },
+              UseInsecureConnection = insecureConnection,
+            };
+
+            var locator = new Locator(locatorHost, locatorPort, locatorParameters);
+            var connectionParameters = CreateConnectionParameters();
+            using (var connectionFuture = locator.ConnectAsync(connectionParameters))
+            {
+                var connectionOption = connectionFuture.Get(5000 /* Using milliseconds */);
+                if (connectionOption.HasValue)
+                {
+                    var connection = connectionOption.Value;
+                    Console.WriteLine("connected: {0}", connection.IsConnected);
+                }
+                else
```

What is this case anyway? Did it indicate the connect has failed?

---

Am not sure TBH, the future just returns no value. I think this swallows errors potentially.